master/internal/api_experiment.go

```diff
@@ -833,6 +856,35 @@ func (a *apiServer) ActivateExperiment(
 	}
 }
 
+func (a *apiServer) ActivateExperiments(
+	ctx context.Context, req *apiv1.ActivateExperimentsRequest,
+) (*apiv1.ActivateExperimentsResponse, error) {
+	query, err := a.editableExperimentIds(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	expIDs := []int32{}
+	err = query.Where("id IN (?)", bun.In(req.ExperimentIds)).
+		Model(&expIDs).
+		Scan(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	resp := &apiv1.ActivateExperimentsResponse{ExperimentIds: []int32{}}
+	mockResp := &apiv1.ActivateExperimentResponse{}
+	for _, expID := range expIDs {
+		addr := experimentsAddr.Child(expID)
+		err = a.ask(addr, &apiv1.ActivateExperimentRequest{Id: expID}, mockResp)
```

Comment:
- it seems the error processing mechanics of `a.ask` aren't really used here, because err is disregarded.
- actor system has `AskAll` which allows asking multiple actors concurrently

similar in spirit to [the use of ctx.GetAll in RPs](https://github.com/determined-ai/determined/blob/master/master/internal/rm/agentrm/resource_pool.go#L794), we could do something along the lines of:

```suggestion
    refs := []*actor.Ref{}
	for _, expID := range expIDs {
		addr := experimentsAddr.Child(expID)
		ref := a.m.system.Get(addr)
		if ref == nil {
		    // error
	    }
	    refs = append(refs, ref)
    }
    // id in the request message doesn't really have to contain experiment id - experiment actor should not care.
    resps := a.m.system.AskAll(&apiv1.ActivateExperimentRequest{}, refs)
    // process resps
```

parts of this will be repeated for `PauseExperiments` and others, so can be refactored out into a small helper.

Reply:
on `resps := a.m.system.AskAll(&apiv1.ActivateExperimentRequest{}, refs)` I get

`cannot use refs (variable of type []*actor.Ref) as type *actor.Ref in argument to a.m.system.AskAll`

Reply:
I think it can be `refs...` and then I just need to debug the resps processing

Reply:
Update: with
```go
	resps := a.m.system.AskAll(&apiv1.PauseExperimentRequest{}, refs...)
	for msg := range resps {
		switch msg.(type) {
		case error:
			return nil, msg.(error)
		default:
			logrus.Infof("succeeded to pause experiment")
		}
	}
```

It's not catching the error for an experiment in state that can't be paused or activated. Also not sure how to get the expID to filter to only successful ones.

Reply:
>It's not catching the error for an experiment in state that can't be paused or activated.

hmm, how does it work for a single experiment / single actor ask?

>Also not sure how to get the expID to filter to only successful ones.

See https://github.com/determined-ai/determined/blob/master/master/pkg/actor/util.go#L25
1. you can do `system.AskAll(...).GetAll()` to get a Ref -> response map
2. or call `resp.Source()` directly to get a `Ref`
and then parse experiment id out of map key / resp.Source `.Address()`.